pkg/worker/context.go

```diff
@@ -137,6 +137,9 @@ type hatchetContext struct {
 	indexMu    sync.Mutex
 	listener   *client.WorkflowRunsListener
 	listenerMu sync.Mutex
+
+	streamEventIndex   int64
+	streamEventIndexMu sync.Mutex
```

Comment:
I didn't do this in Python / TS but maybe I should add a lock around the counter 🤔 - was thinking if you're streaming things back concurrently you wouldn't expect any ordering guarantees, but it probably makes sense to try our best anyways. nice idea 👍